src/Psalm/Internal/Analyzer/Statements/Expression/IssetAnalyzer.php

```diff
@@ -30,7 +31,16 @@ public static function analyze(
                 $var_id = '$this->' . $isset_var->name->name;
 
                 if (!isset($context->vars_in_scope[$var_id])) {
-                    $context->vars_in_scope[$var_id] = Type::getMixed();
+                    if (isset($context->vars_in_scope['$this'])) {
```

Comment:
This check is necessary to keep this test passing:
https://github.com/vimeo/psalm/blob/3f52ded9c37a9ab227b8052e132223efd97acc44/tests/TypeReconciliation/ScopeTest.php#L165-L174

This checks if this is inside an object context and dynamic property access is possible.